### PR TITLE
fix: always send a nonce in the auth request MONGOSH-1905

### PR DESCRIPTION
**Pull request by nirinchev**
Needs https://github.com/mongodb-js/devtools-shared/pull/489 for tests to pass.

### CODE REVIEW

**Review by addaleax — APPROVED** [review]

**Comment by addaleax** [comment]
Looks like there's a number of genuine CI failures:

> RPError: nonce mismatch, expected L4O6qClQ-BhlXlTI8ahMBGHDtVgxUMkYTk1AxufX078, got: undefined

**Comment by addaleax** [comment]
(ah, duh, that's probably because of https://github.com/mongodb-js/devtools-shared/pull/489)

**Comment by nirinchev** [comment]
Those errors should be fixed once we update to the new mock oidc provider. The old one ignores the nonce completely, which now fails validation because we're expecting it in the token response.

**Review by addaleax — APPROVED** [review]

**Comment by addaleax** [comment]
@nirinchev Do you want to consider making this an opt-in feature, as mentioned on Slack?

**Comment by nirinchev** [comment]
Yeah, I'll try and investigate how that would look like.

**Comment by addaleax** [comment]
@nirinchev Fwiw, I think we can really make this an option that's parallel to `passIdTokenAsAccessToken`; it's another boolean flag that's off by default and is exposed through the mongosh CLI options and in the Compass connection form.

In this case, I could see the case for passing the `nonce` by default, usually in that case I'd prefer a boolean option that still defaults to `false`, e.g. something that could be called `skipNonceInAuthCodeRequest`.

The alternative is to survey identity providers and confirm that all relevant ones support nonces – that the tests here pass means that at least Entra ID and Okta do, which is a great starting point.

File: src/plugin.spec.ts [hunk 1]
```diff
@@ -77,6 +77,19 @@ async function delay(ms: number) {
   return await new Promise((resolve) => setTimeout(resolve, ms));
 }
 
+function testAuthCodeFlow(
+  fn: (opts: Partial<MongoDBOIDCPluginOptions>) => Mocha.Func
+): void {
+  for (let skipNonceInAuthCodeRequest of [true, false]) {
+    describe(`with skipNonceInAuthCodeRequest: ${skipNonceInAuthCodeRequest}`, function () {
```
Comment (addaleax) [inline src/plugin.spec.ts:84]: Linting seems to have some comments here

**Review by addaleax — APPROVED** [review]

**Pull request merged**